### DISCUSSION TITLE
如何避免输出字串在 MS Word 中被默认刷成中文字体？

### DISCUSSION THREAD

**Discussion by shenyileirob** [rime/home#1673] [Q&A]
Windows 系统中用小狼毫在 MS Word 中打字，输出字串中部分码位的字符会被默认标记为“简体中文”并默认获得简体中文字体而需要手动更改，这在涉及 IPA、扩展拉丁、基里尔、希腊等文种时特别让人抓狂。是否有可能：

- 下策：将小狼毫附加注册到某西文语言（如英语（美国））下。（我记得这一条很多年之前就有人提出过？）
- 中策：允许在输入方案中指定本输入方案输出字串至少在 MS Word 中被标记为某西文语言。
- 上策：允许在输入方案中指定本输入方案某些码位的字符被标记为西文，或者增加可控制输出语言标记的开关。

**Comment by cqjjjzr** [comment]
同问，因为经常用 rime-kagiroi 打日文，会很抓狂。

按理说因为现代的 RIME 小狼毫用的是 TSF，是能给每个串单独指定语言的（从 TSF 吐出来的文本是一段 `ITfRange`），但恐怕 RIME 核心部分并没有支持给吐出来的串添加这样的支持。

**Comment by shenyileirob** [comment]
将小狼毫手动注册到其他文种下的指引：

1. 打开注册表编辑器，定位到[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\CTF\TIP\{A3F4CDED-B1E9-41EE-9CA6-7B4D0DE6CB0A}\LanguageProfile]，导出。
2. 用记事本阅读文件，注意到文件分5节重复，每节之间唯一的不同是键盘标识符：0x00000404台湾中文，0x00000804大陆中文，0x00000c04香港中文，0x00001004新加坡中文，0x00001404澳门中文。
3. 复制一节粘贴在文件末尾，语言改为0x00000408即为希腊文，或0x00000439即为天城文，等等。完整列表可见 https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/windows-language-pack-default-values?view=windows-11
4. 保存文件。退出后用注册表编辑器打开该文件即可将修改内容导入。现在可以在目标文种选项下找到小狼毫的选项了。
5. 注意部分拉丁系文字（包括英文）并不支持任何输入法，此方法无效。

当然这不是正经的解决方案。我们仍在等待开发者的回应。

**Reply by cqjjjzr** [reply]
我自己 vibe 出了一个方案通过修改 ITfRange 的语言实现，请参考 https://github.com/rime/weasel/pull/1837

但 merge 估计遥遥无期
